src/shared/sol-iio.c

```diff
@@ -0,0 +1,1094 @@
+        trigger_dir);
+    if (!result) {
+        SOL_WRN("Could not open IIO trigger (%s) name on sysfs", trigger_dir);
+        return false;
+    }
+
+    len = sol_util_read_file(path, "%ms", &name);
+    if (len < 0) {
+        SOL_WRN("Could not read IIO trigger (%s) name on sysfs", trigger_dir);
+        return false;
+    }
+
+    result = streq(trigger_name, name);
+    free(name);
+
+    return result;
+}
+
+static bool
+check_manual_triggering(struct sol_iio_device *device)
+{
+    DIR *dir;
+    struct dirent *ent, *res;
+    int success;
+    long name_max;
+    size_t len;
+
+    /* The only way I've found to relact trigger name and trigger is
+     * by opening all triggers on /sys/bus/iio/devices
+     * and checking name by name =/ */
+
+    dir = opendir(SYSFS_DEVICES_PATH);
+    if (!dir) {
+        SOL_WRN("No IIO devices available");
+        return false;
+    }
+
+    /* See readdir_r(3) */
+    name_max = pathconf(SYSFS_DEVICES_PATH, _PC_NAME_MAX);
+    if (name_max == -1)
+        name_max = 255;
+    len = offsetof(struct dirent, d_name) + name_max + 1;
+    ent = malloc(len);
+    SOL_NULL_CHECK(ent, false);
+
+    success = readdir_r(dir, ent, &res);
+    while (success == 0 && res) {
+        if (strstartswith(ent->d_name, "trigger")) {
+            if (check_trigger_name(ent->d_name, device->trigger_name)) {
+                device->manual_triggering = check_file_existence(
+                    SYSFS_TRIGGER_NAME_PATH, ent->d_name);
+                /* triggers dirs are of the form triggerX, so here we save X */
+                device->trigger_id = atoi(ent->d_name + strlen("trigger"));
+                return true;
+            }
+        }
+
+        success = readdir_r(dir, ent, &res);
+    }
+
+    return false;
+}
+
+static bool
+set_current_trigger(struct sol_iio_device *device, const char *trigger_name)
+{
+    char path[PATH_MAX];
+    bool r;
+
+    r = craft_filename_path(path, sizeof(path), CURRENT_TRIGGER_DEVICE_PATH,
+        device->device_id);
+    if (!r) {
+        SOL_WRN("Could not open device current_trigger file");
+        return false;
+    }
+
+    if (sol_util_write_file(path, "%s", trigger_name) < 0) {
+        SOL_WRN("Could not write to device current_trigger file");
+        return false;
+    }
+
+    return true;
+}
+
+static bool
+create_sysfs_trigger(struct sol_iio_device *device)
+{
+    int len, i;
+    char *trigger_name;
+    bool result = false;
+
+    /* First, check if 'iio_sysfs_trigger' is available */
+    if (!check_file_existence(SYSFS_TRIGGER_SYSFS_ADD_TRIGGER)) {
+        SOL_WRN("No 'iio_sysfs_tigger' under '/sys/bus/iio/devices'."
+            " Missing 'modprobe iio-trig-sysfs'?");
+        return false;
+    }
+
+    /* TODO check available trigger indice instead of using a random one */
+    /* Create new trigger */
+    i = rand();
+    if (sol_util_write_file(SYSFS_TRIGGER_SYSFS_ADD_TRIGGER, "%d", i) < 0)
```

Comment:
I recall @lpereira said to just write to the `SYSFS_TRIGGER_SYSFS_ADD_TRIGGER` without prior `check_file_existence()`. To print the same message you can check for `-ENOENT`.